webapp/channels/src/components/initial_loading_screen/initial_loading_screen_class.ts

```diff
@@ -124,6 +125,12 @@ export class InitialLoadingScreenClass {
 
         this.loadingScreenElement.className = LOADING_COMPLETE_CLASS_FOR_SCREEN;
         this.loadingAnimationElement.className = LOADING_COMPLETE_CLASS_FOR_ANIMATION;
+
+        measureAndReport({
+            name: Measure.SplashScreen,
+            startMark: 0,
+            canFail: false,
+        });
```

Comment:
The loading screen currently stays visible for longer when you're loading into a channel because we want to hide as elements are popping in (https://github.com/mattermost/mattermost/commit/5d7aa17a27b8d1e5fbb7be4618a184a630e72db6#diff-6eb34aba3c8a845178ffb6988b166d9f2f68595e04efff413b4f509c64a0307d).

Should we maybe add a label to this to differentiate between the nicer/slower loading screen in the channel vs the faster one in other parts of the app?

Reply:
That diff just points to a big commit for me. Could you clarify what you want to do? A channel will always get loaded when we open Mattermost right? What is the "other parts of the app" case?

Reply:
By other areas of the app, I mean areas like the system console or login page. The regular channel view with the post list and such is under the `TeamController` which keeps the loading screen visible for longer, so this time would be slightly longer for it.

Technically, the loading screen gets hidden when the `TeamController` mounts for the routes under it, but for any other route, the loading screen gets hidden by the `Root` component once the current user and the config are loaded. The `TeamController` is always mounted after the current user and config are loaded, so that would happen later

Reply:
So there are only 2 variations right - stopping from root and stopping from teamController?

```
channels/src/components/root/root.tsx:                InitialLoadingScreen.stop();
channels/src/components/team_controller/team_controller.tsx:        InitialLoadingScreen.stop();
```

Adding a label to differentiate between the two should be sufficient?

Reply:
Yep, that's right. We may add more in the future, but I doubt that'll be any time soon as initial load performance isn't a current focus